fem/bilinearform.hpp

```diff
@@ -705,6 +705,8 @@ class MixedBilinearForm : public Matrix
 
    void Assemble(int skip_zeros = 1);
 
+   void AssembleDiagonal_ADAt(const Vector &D, Vector &diag) const;
```

Comment:
@YohannDudouit I added a comment describing what the function does, but AssembleDiagonal has an additional remark about AMR meshes which I do not understand and do not know whether it applies to this function as well.

Reply:
I'm really not convinced that this method should be part or the interface of a mixed bilinear form, @v-dobrev shouldn't we make this a function instead of a method?